### PR TITLE
Bug 1952667: Update Readme with detailed information about the CBO

### PR DESCRIPTION
**Pull request by sadasu**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,36 @@
...
+
+Introduction
+
+OpenShift supports multiple cloud platform types like AWS, GCP, Azure etc. In addition to these public cloud platforms, OpenShift can also be used to set up a fully functional OpenShift cluster with just Baremetal servers which are either pre-provisioned or are fully provisioned using OpenShift’s Baremetal IPI solution.
+The CBO fits into this specific solution by deploying on an OpenShit control plane all the necessary components required to take an unprovisioned server to a fully functional worker node ready to run OpenShift compute workloads. This second level operator was written based on enhancement :
+https://github.com/openshift/enhancements/blob/master/enhancements/baremetal/an-slo-for-baremetal.md for details about this operator.
+
+What does it do?
+
+Cluster-baremetal-operator (CBO) is designed to be an OpenShfit Operator that is responsible for deploying all components required to provision Baremetal servers into worker nodes that join an OpenShift cluster.
```
Comment (imain) [inline README.md:11]: s/OpenShfit/OpenShift

File: README.md [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+
+What are its inputs?
+
+The CBO needs information about the network to which the baremetal servers are connected and where it can find the image required to boot the servers. This information is provided by the Provisioning CR. CBO watches this resource and passes this information to the metal3 deployment after validating its contents.
+The Provisioning CR with details about the config items can be found here. 
+
+What are its outputs?
+
+If not already created by an external entity, the metal3 deployment and its associated Secrets are the result of the CBO. The CBO also creates an image-cache Daemonset that assists the metal3 deployment by downloading the image provided in the Provisioning CR and making it available locally on each master node so that the metal3 deployment would be able to access a local copy of the image while trying to boot a baremetal server.
```
Comment (imain) [inline README.md:31]: s/would be able to/is able to/

File: README.md [hunk 3]
```diff
@@ -1,3 +1,36 @@
...
+
+CBO reports its own state using the “baremetal” CO as mentioned earlier. It is also designed to provide alerts and metrics regarding its own deployment. It is also capable of reporting metrics gathered by BMO regarding the baremetal servers being provisioned. These metrics can then be scraped by Prometheus and can be viewed on the Prometheus dashboard.
+
+CBO will also start having to report on its own health. That feature currently does not exist and is expected to be implemented in the near future.
```
Comment (imain) [inline README.md:35]: I would guess we want to leave this out until the change is made?  Seems odd, especially as a final sentence in the document.

**Review by imain — COMMENTED** [review]
Looks like a great start Sandhya!

File: README.md [hunk 4]
```diff
@@ -1,3 +1,36 @@
...
+CBO is considered a second level operator (SLO) in OpenShift parlance. What that means is that another OpenShift operator is responsible for deploying CBO. In this case, the Cluster Version Operator (CVO) is responsible for deploying its SLOs at a specific run level that is coded into the manifests of that operator.
+The OpenShift Installer is responsible for deploying the control plane but does not wait for CVO to complete deployment of the CBO. The CVO, also running on the control plane, is completely responsible for running the CBO on one master node at a time. The worker deployment is completely handled by metal3 which in turn deployed by CBO as mentioned earlier.
+
+What are its inputs?
+
+The CBO needs information about the network to which the baremetal servers are connected and where it can find the image required to boot the servers. This information is provided by the Provisioning CR. CBO watches this resource and passes this information to the metal3 deployment after validating its contents.
+The Provisioning CR with details about the config items can be found here.
```
Comment (imain) [inline README.md:27]: I think here we can document the provisioning CR and talk about network settings.  I'll start working on that.

File: README.md (hunk 1 above)
Comment (asalkeld) [reply README.md:11]: yeah and on line 6 check that spelling too :-O

File: README.md [hunk 5]
```diff
@@ -1,3 +1,36 @@
...
+The CBO fits into this specific solution by deploying on an OpenShit control plane all the necessary components required to take an unprovisioned server to a fully functional worker node ready to run OpenShift compute workloads. This second level operator was written based on enhancement :
+https://github.com/openshift/enhancements/blob/master/enhancements/baremetal/an-slo-for-baremetal.md for details about this operator.
+
+What does it do?
+
+Cluster-baremetal-operator (CBO) is designed to be an OpenShfit Operator that is responsible for deploying all components required to provision Baremetal servers into worker nodes that join an OpenShift cluster. 
+The components that have knowledge of how to connect and boot a Baremetal Server are encapsulated in an upstream K8s project called metal3.io. The CBO is responsible for making sure that the metal3 deployment consisting of the baremetal-operator (BMO) and Ironic containers is always running on one of the mater nodes within the OpenShift cluster.
```
Comment (asalkeld) [inline README.md:12]: s/mater/master

File: README.md (hunk 1 above)
Comment (sadasu) [reply README.md:11]: Thanks! Fixed!

File: README.md (hunk 2 above)
Comment (sadasu) [reply README.md:31]: Done.

File: README.md (hunk 3 above)
Comment (sadasu) [reply README.md]: I agree. Removed

**Comment by stbenjam** [comment]
Only editing the readme, we can override it all

/override ci/prow/e2e-metal-ipi-serial-ipv4
/override ci/prow/e2e-metal-ipi-ovn-ipv6 
/override ci/prow/e2e-agnostic
/lgtm

File: README.md [hunk 6]
```diff
@@ -1,3 +1,85 @@
...
+own state via the “baremetal” ClusterOperator resource as is required by every OpenShift operator.
+The CBO reads, validates and passes information provided in the [Provisioning Config Resource (CR)](https://github.com/openshift/cluster-baremetal-operator/blob/master/config/crd/bases/metal3.io_provisionings.yaml) and passes this information to the metal3 deployment. It also creates Secrets that
+containers within the metal3 deployment use to communicate with each other. Currently, only one copy of the Provisioning CR exists per OpenShift
+Cluster so all worker nodes would be provisioned using the same configuration.
+
+
+## When is CBO active?
+
+CBO runs on all platform types supported by OpenShift but will perform its above mentioned tasks only when the platform type is BareMetal. The
+“baremetal” ClusterOperator (CO) displays the current state of the CBO when running on a BareMetal platform and as “Disabled” in all Platform types.
```
Comment (lshilin-r) [inline README.md:28]: May be better "...and as “Disabled” in all other Platform types"?

**Pull request merged**